lib/src/util/unicode_utils.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-42574
+/// https://blog.rust-lang.org/2021/11/01/cve-2021-42574.html
+/// U+202A, U+202B, U+202C, U+202D, U+202E, U+2066, U+2067, U+2068, U+2069.
+var _unsafe = const [
+  '\u202A',
+  '\u202B',
+  '\u202C',
+  '\u202D',
+  '\u202E',
+  '\u2066',
+  '\u2067',
+  '\u2068',
+  '\u2069',
+].map((u) => u.codeUnitAt(0));
```

Comment:
This is a mapped list iterable which will evaluate the `(u) => ...` function for each element for each code unit that is passed to `unsafe`.
So add a `.toList()` or `.toSet()`.

You could use a set literal:
```dart
final _unsafe = { 0x202A, 0x202B, .... };
```
or
```
final _unsafe = {..."\u202A\u202B\u202C\u...".codeUnits};
```

The fastest would probably be to do a range check:
```dart
bool unsafe(int codeUnit) =>
    0x202A <= codeUnit && codeUnit <= 0x202E ||
    0x2066 <= codeUnit && codeUnit <= 0x2069;
```

It is probably also fast to find with a RegExp `[\u202A-\u202E\u2066-\u2069]`.
You could step through the matches with `.indexOf(pattern, start)` - that would give you the code-unit index which presumably can be converted into a line/column.

Reply:
> This is a mapped list iterable  

😬 

Wow.  This is hilarious especially given that I've been thinking about analyses to capture **exactly** this kind of misuse of a lazy iterable.  (See: https://github.com/dart-lang/linter/issues/855.)

> The fastest would probably be to do a range check.

💯 

> You could step through the matches with ...

Oh.  This is a great idea.

Thank you!